test/unit/orchestration/dhcp_test.rb

```diff
@@ -185,4 +185,15 @@ def teardown
     assert_equal 1, bmc.queue.items.select {|x| x.action == [ bmc,     :set_dhcp ] }.size
     assert_equal 1, bmc.queue.items.select {|x| x.action == [ bmc.old, :del_dhcp ] }.size
   end
+
+  test "new host with dhcp and no operating system should show correct validation on save" do
+    h = FactoryGirl.build(:host, :with_dhcp_orchestration)
+    h.operatingsystem = nil
```

Comment:
Would you mind changing those to `operatingsystem.try(:ptables).try(:first)`?  It might help us in the future!